File: part3/cpp/simple/hello_server.cpp
#include "gen-cpp/helloSvc.h"
#include <thrift/server/TSimpleServer.h>
#include <thrift/transport/TServerSocket.h>
#include <thrift/transport/TBufferTransports.h>
#include <thrift/protocol/TBinaryProtocol.h>
#include <boost/make_shared.hpp>
#include <boost/shared_ptr.hpp>
#include <iostream>

using namespace ::apache::thrift::server;
using namespace ::apache::thrift::protocol;
using namespace ::apache::thrift::transport;
using boost::shared_ptr;
using boost::make_shared;

class helloSvcHandler : public helloSvcIf {
public:
    virtual void getMessage(std::string& _return, const std::string& name) override {
        std::cout << "Server received: " << name << ", from client" << std::endl;
        _return = "Hello " + name;
    }
};

int main() {
    auto handler = make_shared<helloSvcHandler>();
    auto proc = make_shared<helloSvcProcessor>(handler);
    auto trans_svr = make_shared<TServerSocket>(9090);
    auto trans_fac = make_shared<TBufferedTransportFactory>();
    auto proto_fac = make_shared<TBinaryProtocolFactory>();
    TSimpleServer server(proc, trans_svr, trans_fac, proto_fac);
    server.serve();
    return 0;
}

File: part3/java/simple/HelloServer.java
import org.apache.thrift.TProcessor;
import org.apache.thrift.server.TSimpleServer;
import org.apache.thrift.transport.TServerSocket;
import org.apache.thrift.transport.TTransportException;
import org.apache.thrift.TException;

public class HelloServer {

    public static class MessageHandler implements helloSvc.Iface {

        @Override
        public String getMessage(String name) throws TException {
            System.out.println("[Server] received: " + name);
            return "Hello " + name;
        }
    }

    public static void main(String[] args) throws TTransportException {
        TServerSocket trans_svr = new TServerSocket(9090);
        TProcessor proc = new helloSvc.Processor<>(new MessageHandler());
        TSimpleServer server = 
            new TSimpleServer(
                new TSimpleServer.Args(trans_svr)
                    .processor(proc)
            );
        System.out.println("[Server] waiting for connections");
        server.serve();
    }
}
